### PR TITLE
Add option to disable edit mode for inline edit widget and disable where necessary

### PR DESCRIPTION
**Pull request by sravfeyn**
cc @pr33thi 

https://dimagi-dev.atlassian.net/browse/HI-491

### CODE REVIEW

File: corehq/apps/app_manager/templates/app_manager/app_view.html [hunk 1]
```diff
@@ -52,6 +52,7 @@
         rows: 1,
         saveValueName: 'name',
         nodeName: 'input',
+        allow_edit: '{{ request.couch_user.can_edit_apps|yesno:"true,false" }}',
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/app_view.html]: Expecting "true" or "false" as a string value is confusing, can this accept real booleans instead?

File: corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js [hunk 2]
```diff
@@ -7,6 +7,7 @@
  *  - value: Text to display and edit
  *  - name: HTML name of input
  *  - id: HTML id of input
+ *  - allow_edit: Whether the edit mode should be allowed or not
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js]: Given that this widget's purpose is to edit, I'd expect disabling it to be an unusual case, so for this to be an optional param and to be the inverse, something like `disable` or `disallow_edit`.

**Comment by orangejenny** [comment]
It'd be nice if preventing editing also turned off the highlight on hover and the hand cursor, since those make it seem like clicking on the value should do something.

File: corehq/apps/app_manager/templates/app_manager/app_view.html (hunk 1 above)
Comment (sravfeyn) [reply corehq/apps/app_manager/templates/app_manager/app_view.html]: Sure, there was some issue in the beginning, but made it a boolean now.

File: corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js (hunk 2 above)
Comment (sravfeyn) [reply corehq/apps/hqwebapp/static/hqwebapp/js/components/inline_edit.js]: Updated.

**Review by orangejenny — APPROVED** [review]
Thanks for the quick updates, looks good.

**Pull request merged**